src/pedersen.rs

```diff
@@ -1453,15 +1452,16 @@ mod tests {
 		// Ensure rewinding works
 
 		let blinding = SecretKey::new(&secp, &mut thread_rng());
-		let nonce = SecretKey::new(&secp, &mut thread_rng());
+		let rewind_nonce = SecretKey::new(&secp, &mut thread_rng());
+		let private_nonce = SecretKey::new(&secp, &mut thread_rng());
 		let value = 12345678;
 		let commit = secp.commit(value, blinding.clone()).unwrap();
 
 		let bullet_proof =
-			secp.bullet_proof(value, blinding.clone(), nonce.clone(), Some(extra_data.clone()), None);
+			secp.bullet_proof(value, blinding.clone(), private_nonce.clone(), private_nonce.clone(), Some(extra_data.clone()), None);
```

Comment:
This tests uses the same nonce for both on purpose. The rewind only assumes that for both `mu` and `tau_1`, `tau_2` the same nonce was used. If we want to test that it reproduces both the value and the blinding factor, we have to use the same nonce for now. We can add a more sophisticated rewind function later, but since we don't need it in the wallet we can save that for after the HF.
This test is also useful since it mimics the legacy bulletproof scheme.